tubular/nominal.py

```diff
@@ -343,7 +349,71 @@ def __init__(
 
         self.unseen_levels_to_rare = unseen_levels_to_rare
 
-    def fit(self, X: pd.DataFrame, y: pd.Series | None = None) -> pd.DataFrame:
+    @nw.narwhalify
+    def _check_strlike_columns(self, X: FrameT) -> None:
```

Comment:
N: would put an underscore between str_like personally.  Had to double take when I first read it as one word.